src/main/java/com/sap/oss/phosphor/fosstars/model/score/oss/OssArtifactSecurityScore.java

```diff
@@ -0,0 +1,111 @@
+import com.sap.oss.phosphor.fosstars.model.value.ScoreValue;
+import com.sap.oss.phosphor.fosstars.model.value.ValueHashSet;
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * <p>This is a security scoring function for open-source artifact versions.
+ * The scoring function is based on the following sub-scores:</p>
+ * <ul>
+ *   <li>{@link ArtifactVersionSecurityScore}</li>
+ *   <li>{@link OssSecurityScore}</li>
+ * </ul>
+ */
+public class OssArtifactSecurityScore extends AbstractScore {
+
+  /**
+   * A description of the score.
+   */
+  private static final String DESCRIPTION
+      = "The security score for open source artifact versions shows how safe it is to use a "
+      + "specific version of an open source component. Besides information about the artifact "
+      + "version, the scoring functions takes into account the security score of the project "
+      + "to which the artifact belongs to.";
+
+
+  private final ArtifactVersionSecurityScore artifactVersionSecurityScore;
+  private final OssSecurityScore ossSecurityScore;
+
+  /**
+   * Initialize the score with default values.
+   */
+  public OssArtifactSecurityScore() {
+    this(new ArtifactVersionSecurityScore(), new OssSecurityScore());
+  }
+
+  /**
+   * Initializes a new security score for open-source artifacts.
+   *
+   * @param artifactVersionSecurityScore artifact version and releases based score
+   * @param ossSecurityScore project specific security score
+   */
+  @JsonCreator
+  public OssArtifactSecurityScore(@JsonProperty("artifactVersionSecurityScore")
+      ArtifactVersionSecurityScore artifactVersionSecurityScore,
+      @JsonProperty("ossSecurityScore") OssSecurityScore ossSecurityScore) {
+
+    super("Security score for an artifact of an open-source project", DESCRIPTION);
+    this.artifactVersionSecurityScore = artifactVersionSecurityScore;
+    this.ossSecurityScore = ossSecurityScore;
+  }
+
+  @Override
+  public Set<Feature<?>> features() {
+    return Collections.emptySet();
+  }
+
+  @Override
+  public Set<Score> subScores() {
+    return setOf(artifactVersionSecurityScore, ossSecurityScore);
+  }
+
+  @Override
+  public ScoreValue calculate(Value<?>... values) {
+    ScoreValue artifactVersionScore = calculateIfNecessary(
+        new ArtifactVersionSecurityScore(), new ValueHashSet(values));
+    ScoreValue projectSecurityScore = calculateIfNecessary(
+        new OssSecurityScore(), new ValueHashSet(values));
+
+    ScoreValue scoreValue = scoreValue(Score.MIN, artifactVersionScore, projectSecurityScore);
+
+    if (artifactVersionScore.isUnknown()) {
+      return scoreValue.withMinConfidence().makeUnknown();
+    }
+
+    final double updatedScore;
+    if (artifactVersionScore.get() >= 9.0) {
+      updatedScore = 0.11 * artifactVersionScore.get() * projectSecurityScore.get();
+    } else {
+      updatedScore = 0.1 * (1 + artifactVersionScore.get()) * projectSecurityScore.get();
+    }
```

Comment:
Have added a test and added a check in case that final score is greater than max score.
During this I realised that the `ScoreValue.set(..)` method also check that no score is greater than max score.
Hence, I'm not sure if I understand your hint regarding "_score value overflow._" correctly.
Is it required that I do `return scoreValue.set(Math.min(Score.MAX, updatedScore));` or should this be reverted?

Reply:
I mean it would be good tests with extreme conditions for the above formulas. `updatedScore` should not be more than `Score.MAX`.

Reply:
I mean it would be good tests with extreme conditions for the above formulas. `updatedScore` should not be more than `Score.MAX`.

Reply:
Have added a test `testMaxScore()` to verify that resulting score is not above `Score.MAX`.

But I'm not sure about the part regarding the formulas.
The idea is to use the project security score as base and increase it slightly if the artifact version security score is above 9 and decrease it if it is below 9.
As result it may happen that in theory the project score is 10 and the artifact score is 10 which results in a `updatedScore` of 11 which is adjusted to the `Score.MAX`.
To change the formula to never result in a score higher than `Score.MAX` but keep the intension may be a bit tricky.

Reply:
> To change the formula to never result in a score higher than Score.MAX but keep the intension may be a bit tricky.

You can add a check if `updatedScore` is less than `Score.MAX`, or you can use `Score.adjust()`.

Reply:
Done 👍